recipes/security/api-umbrella/readme.md

```diff
@@ -0,0 +1,469 @@
+nero0vahaa8h        mongo-rs_controller.1                      martel/mongo-replica-ctrl:latest   ms-manager0         Running             Running 5 seconds ago
+```
+
+Set the connection url for mongo based on the IPs of your Swarm Cluster:
+
+```
+  $ MONGO_REPLICATE_SET_IPS=192.168.99.100:27017,192.168.99.101:27017,192.168.99.102:27017
+  $ export MONGO_REPLICATE_SET_IPS
+```
+
+If you used `miniswarm` to create your cluster, you can get the different IPs
+using the `docker-machine ip` command, e.g.:
+
+```
+    $ docker-machine ip ms-manager0
+    
+    $ docker-machine ip ms-worker0
+    
+    $ docker-machine ip ms-worker1
+```
+
+when all services will be in status ready, your backend is ready to be used:
+
+```
+    $ sh deploy_front.sh
+
+    generating config file
+    replacing target file  api-umbrella.yml
+    replace mongodb with mongo-rs_mongo
+    replacing target file  api-umbrella.yml
+    replace rs_name with rs
+    Creating config api_api-umbrella
+    Creating service api_api-umbrella
+```
+
+When also the frontend services will be running, your deployment
+will look like this:
+
+```
+    $ docker service ls
+ID                  NAME                  MODE                REPLICAS            IMAGE                                 PORTS
+ca11lmx40tu5        api_api-umbrella      replicated          2/2                 martel/api-umbrella:0.14.4-1-fiware   *:80->80/tcp,*:443->443/tcp
+te1i0vhwtmnw        mongo-rs_controller   replicated          1/1                 martel/mongo-replica-ctrl:latest      
+rbo2oe2y0d72        mongo-rs_mongo        global              3/3                 mongo:3.2
+```
+
+If you see `3/3`in the replicas column it means the 3 out of 3 planned replicas
+are up and running.
+
+## A walkthrough
+
+Now, let's create the admin user in API Umbrella. As first thing,
+get the IP of your master node:
+
+```
+    $ docker-machine ip ms-manager0
+```
+
+Open the browser at the following endpoint: [http://ip/admin](http://ip/admin).
+
+Unless you also created certificates for your server, API Umbrella,
+will ask you to accept the connection to an insecure instance.
+
+In the page displayed you can enter the admin user name and the password.
+
+Now you are logged in and you can configure the backend APIs.
+
+In the menu select `Users->Admin Accounts` and click on the username you just
+created. Copy the `Admin API Access` for your account.
+
+In the menu select "Users->Api Users" click on the username
+`web.admin.ajax@internal.apiumbrella` and copy the Api
+Key (of course you can create new ones instead of reusing API Umbrella
+defaults).
+
+Create a simple API to test that everything works:
+
+```
+$ curl -k -X POST "https://<your-cluster-manager-ip>/api-umbrella/v1/apis" -H "X-Api-Key: <your-API-KEY>" -H "X-Admin-Auth-Token: <your-admin-auth-token>" -H "Accept: application/json" -H "Content-Type: application/json" -d @- <<EOF
```

Comment:
I'd suggest placing them in files next to readme.

Reply:
not sure what you mean